roqoqo_qiskit_devices/src/devices/ibm_belem.rs

```diff
@@ -0,0 +1,439 @@
+                    .unwrap();
+            }
+        }
+
+        device
+    }
+
+    /// Returns the IBM's identifier.
+    ///
+    /// # Returns
+    ///
+    /// A str of the name IBM uses as identifier.
+    pub fn name(&self) -> &'static str {
+        "ibmq_belem"
+    }
+}
+
+impl Default for IBMBelemDevice {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl From<&IBMBelemDevice> for IBMDevice {
+    fn from(input: &IBMBelemDevice) -> Self {
+        Self::IBMBelemDevice(input.clone())
+    }
+}
+
+impl From<IBMBelemDevice> for IBMDevice {
+    fn from(input: IBMBelemDevice) -> Self {
+        Self::IBMBelemDevice(input)
+    }
+}
+
+impl IBMBelemDevice {
+    /// Setting the gate time of a single qubit gate.
+    ///
+    /// # Arguments
+    ///
+    /// * `gate` - hqslang name of the single-qubit-gate.
+    /// * `qubit` - The qubit for which the gate time is set.
+    /// * `gate_time` - gate time for the given gate.
+    pub fn set_single_qubit_gate_time(
+        &mut self,
+        gate: &str,
+        qubit: usize,
+        gate_time: f64,
+    ) -> Result<(), RoqoqoError> {
+        if qubit >= self.number_qubits {
+            return Err(RoqoqoError::GenericError {
+                msg: format!(
+                    "Qubit {} larger than number qubits {}",
+                    qubit, self.number_qubits
+                ),
+            });
+        }
+        match self.single_qubit_gates.get_mut(gate) {
+            Some(gate_times) => {
+                let gatetime = gate_times.entry(qubit).or_insert(gate_time);
+                *gatetime = gate_time;
+            }
+            None => {
+                let mut new_map = HashMap::new();
+                new_map.insert(qubit, gate_time);
+                self.single_qubit_gates.insert(gate.to_string(), new_map);
+            }
+        }
+        Ok(())
+    }
+
+    /// Setting the gate time of a two qubit gate.
+    ///
+    /// # Arguments
+    ///
+    /// * `gate` - hqslang name of the two-qubit-gate.
+    /// * `control` - The control qubit for which the gate time is set.
+    /// * `target` - The target qubit for which the gate time is set.
+    /// * `gate_time` - gate time for the given gate.
+    pub fn set_two_qubit_gate_time(
+        &mut self,
+        gate: &str,
+        control: usize,
+        target: usize,
+        gate_time: f64,
+    ) -> Result<(), RoqoqoError> {
+        if control >= self.number_qubits {
+            return Err(RoqoqoError::GenericError {
+                msg: format!(
+                    "Qubit {} larger than number qubits {}",
+                    control, self.number_qubits
+                ),
+            });
+        }
+        if target >= self.number_qubits {
+            return Err(RoqoqoError::GenericError {
+                msg: format!(
+                    "Qubit {} larger than number qubits {}",
+                    target, self.number_qubits
+                ),
+            });
+        }
```

Comment:
Here we should also add a check that only allows setting the gate time, if the edge is already in the device. So we can prevent the user adding 2qubit connections that are not really available on the physical hardware. This also goes for other devices.